src/Gamepad.js

```diff
@@ -0,0 +1,78 @@
+let gamepadId = 0;
+
+import { GamepadButton } from "./GamepadButton.js";
+
+export class Gamepad extends HTMLDivElement {
+  constructor() {
+    super();
+
+    const id = gamepadId++;
+
+    // setup the gamepad's front end
+    this.innerHTML = `
+      <style>
+        .gamepad {
+          border: 1px solid grey;
+          display: inline-block;
+          margin: 5px;
+        }
+      </style>
+      Hello gamepad ${id}!
+    `;
+    this.classList.add("gamepad");
+
+    // set default attributes
+    this._id = id;
+    this._index = 0; // temporary value for now
+    this._connected = false;
+    this._timestamp = new Date().getTime();
+    this._mapping = []; // temporary value for now
+    this._axes = [0.0, 0.0, 0.0, 0.0];
+    this._buttons = gamepadButtons();
+
+    function gamepadButtons() {
+      const buttons = Array(16)
+        .fill(undefined)
+        .map(x => new GamepadButton());
```

Comment:
there is probably a better way to do this. right now, this line results in an `'x' is defined but never used` error. Have yet to figure out how to do this while making sure that each instance of `GamepadButton` is a separate object.

Reply:
We should only add buttons by gamepad controller type (and not restrict to any arbitrary number). So, it might be cool to have:

```JS
/*
Stuff specific for Sony... or "Generic" gamepad? 
*/
SonyGamepad extends GamePad {
   super();
   this._buttons = [
      new GamepadButton(), // top
      new GamepadButton(), // Right 
      // and so on... 
   ]
}; 
```